R/lazy-join-query.R

```diff
@@ -171,7 +173,8 @@ sql_build.lazy_multi_join_query <- function(op, con, ..., sql_options = NULL) {
   op$joins$by <- purrr::map2(
     op$joins$by, seq_along(op$joins$by),
     function(by, i) {
-      by$x_as <- table_names_out[op$joins$by_x_table_id[[i]]]
+      # @mgrlich: x_as should be a single value, right?
```

Comment:
`x_as` can also be multiple values in other cases, e.g.

```r
lf1 <- lazy_frame(x1 = 1)
lf2 <- lazy_frame(x = 1, y2 = 1)
lf3 <- lazy_frame(x = 1, y = 1)

lf1 %>% 
  left_join(lf2, by = c(x1 = "x")) %>% 
  left_join(lf3, by = c(x1 = "x", y2 = "y"))
```

The join columns for `lf3` are from two different tables, `lf1` and `lf2`.

Reply:
When I run that query and look at `ops$join$by` I see this:

```
List of 2
 $ :List of 7
  ..$ x         : chr "x1"
  ..$ y         : chr "x"
  ..$ condition : chr "=="
  ..$ on        : 'sql' chr(0) 
  ..$ na_matches: chr "never"
  ..$ x_as      : 'dbplyr_table_name' chr "`lf1`"
  ..$ y_as      : 'dbplyr_table_name' chr "`lf2`"
 $ :List of 7
  ..$ x         : chr [1:2] "x1" "y2"
  ..$ y         : chr [1:2] "x" "y"
  ..$ condition : chr [1:2] "==" "=="
  ..$ on        : 'sql' chr(0) 
  ..$ na_matches: chr "never"
  ..$ x_as      : 'dbplyr_table_name' chr [1:2] "`lf1`" "`lf2`"
  ..$ y_as      : 'dbplyr_table_name' chr "`lf3`"
```

I'm confused because the `_as` values are table names, not column names, and there's two y's but only one `y_as` in the second join.

Reply:
Should `_as` really be `_table`?

Reply:
And then the other problem is that I broke the vectorisation of `sql_table_prefix()`

Reply:
> I'm confused because the `_as` values are table names, not column names, and there's two y's but only one `y_as` in the second join.

It kind of makes sense that `x_as` and `y_as` are table names as they come from `left_join(..., x_as = ...)`. But in the context of `by` it is indeed a bit confusing.
For `y_as` we don't need more than one table name. The join columns always come from the table `y`.
For `x_as` we might need multiple names as the join columns can come from any of the previous tables joined with (as in this example).
But I think we could make this clearer by renaming to `x_tables` and `y_table` (singular).